conformance/lang/expressions/let-expr/let_expr.balt

```diff
@@ -0,0 +1,431 @@
+    // @output aaa11bbb22ccc
+    io:println(let string v1 = "11", string v2 = "22" in (xml `<!--aaa${v1}bbb${v2}ccc-->`).toString());
+    // @output <!--aaa11bbb22ccc-->
+    io:println(let string v1 = "11", string v2 = "22" in (xml `<?foo aaa${v1}bbb${v2}ccc?>`).toString());
+    // @output <?foo aaa11bbb22ccc?>
+    io:println(let string v1 = "11", string v2 = "22" in (xml `<foo bar="aaa${v1}bb'b${v2}ccc?"/>`).toString());
+    // @output <foo bar="aaa11bb'b22ccc?"></foo>
+    io:println(let int i = 42, float f = 3.14, decimal d = 31.4444, string s = "this-is-a-string",
+    xml elem = xml `<abc/>` in (xml `<elem>${i}|${f}|${d}|${s}|${elem}</elem>`).toString());
+    // @output <elem>42|3.14|31.4444|this-is-a-string|<abc></abc></elem>
+}
+
+Test-Case: output
+Description: Test let-expr, when type-descriptor is a list-type-descriptor and tuple-type-descriptor
+             which are structured-type-descriptor.
+Labels: let-expr, list-constructor-expr, mapping-constructor-expr, function-call-expr
+
+function init() {
+    io:println(let var l = ["A", 2] in l[0]); // @output A
+    io:println(let var l = [["A", 2], ["B", 3, "C"]] in l[1][2]); // @output C
+    io:println(() == let map<any> addressOne = {city: "Colombo", "country": "SriLanka"}, map<any>[] array = []
+        in array.push(addressOne)); // @output true
+    io:println(let int[] arr = [23, 34] in arr.length()); // @output 2
+    io:println(let [int, boolean][3] sealedArray = [[2, true], [3, false], [6, true]]
+        in sealedArray.length()); // @output 3
+    io:println(() == let [int, string...] x = [1, "a", "b"] in x.push("2")); // @output true
+}
+
+Test-Case: output
+Description: Test let-expr, when type-descriptor is a mapping-type-descriptor which is a structured-type-descriptor.
+Labels: let-expr, mapping-constructor-expr
+
+function init() {
+    io:println(let map<any> animals = {"animal1":"Lion", "animal2":"Cat"} in animals.get("animal1")); // @output Lion
+    io:println(let map<any> animals = {"animal1":"Lion", "animal2":"Cat"} in animals.keys());
+    // @output ["animal1","animal2"]
+    io:println(let map<any> animals = {"animal1":"Lion", "animal2":"Cat"} in animals.entries());
+    // @output {"animal1":["animal1","Lion"],"animal2":["animal2","Cat"]}
+    io:println(let string i = "i", float f = 2.0, map<anydata> b = {s: "hello", [i]: 1, f: f}
+        in b["s"] == "hello" && b["i"] == 1 && b["f"] == 2.0); // @output true
+    io:println(let map<any> list1 = {"item1": "item1"} in list1["item1"] is string); // @output true
+    io:println(let map<any> address = {country: "SriLanka"}, map<any> info = {name: "Supun", address: address}
+        in (<map<any>> (info["address"]))["country"]); // @output SriLanka
+}
+
+Test-Case: output
+Description: Test let-expr, when type-descriptor is a record-type-descriptor which is a structured-type-descriptor.
+Labels: let-expr, record-type, mapping-constructor-expr
```

Comment:
Let's add the missing labels in other places too.

Reply:
Added